cmd/infra/powervs/create.go

```diff
@@ -1159,13 +1163,28 @@ func (infra *Infra) setupPowerVSDHCP(ctx context.Context, options *CreateInfraOp
 	return nil
 }
 
+// isTimeoutError validates err is timeout error or not, if yes increase the retryCount, if not return the err
+// retry limit error is thrown when retryCount reaches the limit
+func isTimeoutError(err error, currentRetry *int) error {
```

Comment:
I'm wondering if we can make this generic so that can be used with any API's error..